execution_engine_testing/tests/src/test/system_contracts/auction/distribute.rs

```diff
@@ -652,6 +652,767 @@ fn should_withdraw_bids_after_distribute() {
+        let delegator_stake_before = U512::from(DELEGATOR_1_STAKE);
+        let delegator_stake_after = delegator_1_staked_amount_1;
+
+        delegator_stake_after - delegator_stake_before
+    };
+    let expected_delegator_1_payout_1 =
+        (expected_total_reward_1 * participant_portion).to_integer();
+    assert_eq!(delegator_1_updated_stake_1, expected_delegator_1_payout_1);
+
+    let delegator_2_updated_stake_1 = {
+        let delegator_stake_before = U512::from(DELEGATOR_2_STAKE);
+        let delegator_stake_after = delegator_2_staked_amount_1;
+        delegator_stake_after - delegator_stake_before
+    };
+    let expected_delegator_2_payout_1 =
+        (expected_total_reward_1 * participant_portion).to_integer();
+    assert_eq!(delegator_2_updated_stake_1, expected_delegator_2_payout_1);
+
+    let total_payout_1 =
+        validator_1_updated_stake_1 + delegator_1_updated_stake_1 + delegator_2_updated_stake_1;
+    assert_eq!(total_payout_1, expected_total_reward_1_integer);
+
+    let era_info_1 = {
+        let era = builder.get_era();
+
+        let era_info_value = builder
+            .query(None, Key::EraInfo(era), &[])
+            .expect("should have value");
+
+        era_info_value
+            .as_era_info()
+            .cloned()
+            .expect("should be era info")
+    };
+
+    assert!(matches!(
+        era_info_1.select(*VALIDATOR_1).next(),
+        Some(SeigniorageAllocation::Validator { validator_public_key, amount })
+        if *validator_public_key == *VALIDATOR_1 && *amount == expected_validator_1_payout_1
+    ));
+
+    assert!(matches!(
+        era_info_1.select(*DELEGATOR_1).next(),
+        Some(SeigniorageAllocation::Delegator { delegator_public_key, amount, .. })
+        if *delegator_public_key == *DELEGATOR_1 && *amount == expected_delegator_1_payout_1
+    ));
+
+    assert!(matches!(
+        era_info_1.select(*DELEGATOR_2).next(),
+        Some(SeigniorageAllocation::Delegator { delegator_public_key, amount, .. })
+        if *delegator_public_key == *DELEGATOR_2 && *amount == expected_delegator_2_payout_1
+    ));
+
+    // Next round of rewards
+    for _ in 0..5 {
+        builder.run_auction(timestamp_millis);
+        timestamp_millis += TIMESTAMP_MILLIS_INCREMENT;
+    }
+
+    let distribute_request_2 = ExecuteRequestBuilder::standard(
+        SYSTEM_ADDR,
+        CONTRACT_AUCTION_BIDS,
+        runtime_args! {
+            ARG_ENTRY_POINT => METHOD_DISTRIBUTE,
+            ARG_REWARD_FACTORS => reward_factors,
+        },
+    )
+    .build();
+
+    builder.exec(distribute_request_2).commit().expect_success();
+
+    let total_supply_2 = builder.total_supply(None);
+    assert!(total_supply_2 > initial_supply_1);
+
+    let expected_total_reward_2 = *GENESIS_ROUND_SEIGNIORAGE_RATE
+        * (initial_supply_1
+            + expected_validator_1_payout_1
+            + expected_delegator_1_payout_1
+            + expected_delegator_2_payout_1);
```

Comment:
Maybe
```rust
    let expected_total_reward_2 = *expected_total_reward_1
        + expected_validator_1_payout_1
        + expected_delegator_1_payout_1
        + expected_delegator_2_payout_1;
```

Reply:
No it won't work unfortunately without `GENESIS_ROUND_SEIGNIORAGE_RATE` multiplier but I realized I could simplify this a bit and have `let expected_total_reward_2 = *GENESIS_ROUND_SEIGNIORAGE_RATE * total_supply_2` assuming that `total_supply_2` would be moved before 2nd distribute call

Reply:
Gotcha